External/Plugins/HaXeContext/Linters/DiagnosticsLinter.cs

```diff
@@ -27,7 +27,7 @@ public void LintAsync(IEnumerable<string> files, LintCallback callback)
         {
             var context = ASContext.GetLanguageContext("haxe") as Context;
 
-            if (context == null || !CanContinue(context)) return;
+            if (context == null || !(PluginBase.CurrentProject is ProjectManager.Projects.Haxe.HaxeProject) || !CanContinue(context)) return;
```

Comment:
One would not try to get the context if the project is not haxe. Yes, I understand that this is a miserable optimization, but it seems so much more readable.